### ISSUE TITLE
Error in binding to self when calling a function from a metaclass

### ISSUE THREAD

**Issue by amotzop** [microsoft/pyright#4637]
**Describe the bug**
When trying to call a function of certain metaclasses there is a problem with "binding it to self".
It might be a problem with the typing in typeshed, but I'm not sure.

**To Reproduce**
Run pyright on the code below. The typing was taken from typeshed.

**Expected behavior**
There shoudn't be a warning here as we call a function of a class, so no binding to self needs to happen.

**Screenshots or Code**
```python
from enum import EnumMeta
from typing import TypeVar

_EnumMemberT = TypeVar("_EnumMemberT")

class EnumMeta2(EnumMeta):
    def __getitem__(cls: type[_EnumMemberT], name: str) -> _EnumMemberT:
        return EnumMeta.__getitem__(cls, name)
```
pyright gives the following error message on the last line 
```
Could not bind method "__getitem__" because "Type[EnumMeta]" is not assignable to parameter "self"
  Type "EnumMeta" cannot be assigned to type "Type[_EnumMemberT@__getitem__]"
```

**VS Code extension or command-line**
Happens in both, command-line version 1.1.293

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.294, which I just published. It will also be included in a future release of pylance.
